app/upgrades/v17/erc20_test.go

```diff
@@ -0,0 +1,191 @@
+package v17_test
+
+import (
+	"testing"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	v17 "github.com/evmos/evmos/v16/app/upgrades/v17"
+	erc20precompile "github.com/evmos/evmos/v16/precompiles/erc20"
+	testkeyring "github.com/evmos/evmos/v16/testutil/integration/evmos/keyring"
+	testnetwork "github.com/evmos/evmos/v16/testutil/integration/evmos/network"
+	utiltx "github.com/evmos/evmos/v16/testutil/tx"
+	"github.com/evmos/evmos/v16/utils"
+	"github.com/evmos/evmos/v16/x/erc20/types"
+	evmkeeper "github.com/evmos/evmos/v16/x/evm/keeper"
+	"github.com/stretchr/testify/require"
+)
+
+// SetupNetwork sets up a new test network and returns the network and the grpc handler.
+func SetupNetwork() *testnetwork.UnitTestNetwork {
+	keyring := testkeyring.New(1)
+
+	return testnetwork.NewUnitTestNetwork(
+		testnetwork.WithPreFundedAccounts(keyring.GetAllAccAddrs()...),
+	)
+}
+
+func TestRegisterERC20Extensions(t *testing.T) {
+	var network *testnetwork.UnitTestNetwork
+
+	ibcDenom := utils.ComputeIBCDenom("transfer", "channel-0", "uosmo")
+	ibcTokenPair := types.NewTokenPair(utiltx.GenerateAddress(), ibcDenom, types.OWNER_MODULE)
+	nativeTokenPair := types.NewTokenPair(utiltx.GenerateAddress(), "test", types.OWNER_MODULE)
+	otherNativeTokenPair := types.NewTokenPair(utiltx.GenerateAddress(), "other", types.OWNER_MODULE)
+	externalTokenPair := types.NewTokenPair(utiltx.GenerateAddress(), "uext", types.OWNER_EXTERNAL)
+
+	testcases := []struct {
+		name        string
+		malleate    func()
+		expPass     bool
+		errContains string
+		postCheck   func()
+	}{
+		{
+			name:    "pass - no token pairs in ERC20 keeper",
+			expPass: true,
+			postCheck: func() {
+				requireActiveDynamicPrecompiles(t, network.GetContext(), network.App.EvmKeeper, []string{})
+			},
+		},
+		{
+			name: "pass - native token pair",
+			malleate: func() {
+				network.App.Erc20Keeper.SetTokenPair(network.GetContext(), nativeTokenPair)
+			},
+			expPass: true,
+			postCheck: func() {
+				// Check that the precompile was registered
+				activeDynamicPrecompiles := network.App.EvmKeeper.GetParams(network.GetContext()).ActiveDynamicPrecompiles
+				require.Contains(t, activeDynamicPrecompiles, nativeTokenPair.GetERC20Contract().String(), "expected precompile to be registered")
+
+				// Check that the precompile is set as active
+				requireActiveDynamicPrecompiles(
+					t, network.GetContext(), network.App.EvmKeeper,
+					[]string{nativeTokenPair.Erc20Address},
+				)
```

Comment:
This is not clear to me. 
- The first check is against ActiveDynamicPrecompile, so every dynamic precompile that is dynamic and active.
- The second check is to confirm that the Erc20Contract is active. 
How is it possible to have the first true and the second false?
Is it a wrong naming? Is the the field ActiveDynamicPrecompiles of the param holding all REGISTERED dynamic precompiles and not the ACTIVE ones?

Reply:
ahh good point, this is actually duplicate 👀  thanks for flagging 🙏